### DISCUSSION TITLE
Add a suggestion delay option in the configuration

### DISCUSSION THREAD

**Discussion by tschallacka** [community/community#17670] [Copilot Conversations]
In PHPstorm or Pycharm when I use co-pilot, when I need to press tab I often times inadvertently hit the auto complete whilst I only wanted a tab. Similar as described in https://github.com/github/feedback/discussions/17238

I also have that when I wish to ode something, the suggestion is wrong, but copilot keeps adding it's suggestion , messing up enter presses and tabs I might need, or just plainly distracting me.
Often times copilot also comes with it's wrong suggestion, and the phpstorm autocomplete doesn't work without an explicit alt+space to get the suggestions list.

Please add a setting in the configuration for copilot, suggestion delay for suggestions. Default it can be 0 ms, which people can set to their own liking before they see an copilot option.

And maybe a silence duration option, which will prevent copilot from giving a suggestion for given milliseconds when esc has been pressed to cancel a suggestion. 

It's tedious having to press ESC, then tab to ident, then press ESC again, then tab, etc....

I just want copilot to "shut up" for 5 seconds/configurable time when i've pressed esc.

**Comment by luketych** [comment]
I second this

**Reply by FlinthDevs** [reply]
Same for me ! It's a bit too much in my way on day-to-day tasks, even a fixed delayed value like 500ms would be great !

**Reply by drew3k** [reply]
yeah

**Comment by astratagem** [comment]
This is also an issue in Visual Studio Code. The lack of options for customising the UX is extremely frustrating as an active impediment to Copilot's usability and full integration into one's daily workflow.

In terms of the sheer amount of text Copilot spits into the editor, it is far more verbose than normal completion. The latter has plenty of options for customisation (e.g. delay) and yet Copilot, which is more visually invasive, has nothing. Except for the ability to just… turn it off?

If Copilot were in beta or newly-released, I could understand that its product team would have prioritized the core functionality over usability. But considering how long Copilot has been out, it seems like improving the interface usability is pretty low on their priority list…

**Comment by SnaveSutit** [comment]
In VSC you can press CTRL + space to force show code suggestions. This forcefully closes the copilot suggestion

**Comment by bsaunders27** [comment]
Bump, lack of this feature will probably motivate me to cancel my Copilot subscription, need to be able to set a delay similar to VSCode's TextEditor Intellisense's editor.hover.delay setting (https://github.com/microsoft/vscode/issues/55098). It's an incredibly useful service but just a lot of visual noise when trying to code.

**Reply by Zaid1881** [reply]
- ![dump-stats-2024_1001_231539_stop.txt](https://github.com/user-attachments/files/20536005/dump-stats-2024_1001_231539_stop.txt)

**Comment by Scatterblak** [comment]
Bump again. Why is this still an issue? VERY annoying.

**Comment by FrancoisGrosbellet** [comment]
+1 here, this feature would really be appreciated. The lack of this feature is currently Copilot's biggest issue for my usage.

Very similar request here: [Option to set delay for suggestions would be nice #9978](https://github.com/orgs/community/discussions/9978)

**Comment by felipelalli** [comment]
Duplicated: #9978

**Comment by mpaw** [comment]
It's been two years since the first message in this thread. Any plans to allow us to set a custom delay before copilot provides a suggestion?

**Comment by gerardsimons** [comment]
Yes. Must. Have. I get very annoyed when I am typing a comment and it thinks to know what I am going to type even though I only typed "the" or something like that. How in god's name do you think you know what the comment is going to be about? Superfluous suggestions are very counter-productive, sometimes true intelligence requires you to be silent.

**Reply by MarelPup** [reply]
I 100% agree with [ insert copilot irrelevant suggestion ]

**Comment by jacsamg** [comment]
I came here because I can't stand the instant autocomplete in VScode anymore. I need a delay time, as it's distracting when you're focused.

**Comment by aeckstorm** [comment]
I just don't use it until i need to anymore, because it is outta hand. You can download a status bar toggle to get by until they listen to the community. 

Extension VS Code: copilot-inline-toggle

**Reply by aeckstorm** [reply]
I gave up, switched to windsurf and it's pretty good.

**Reply by coredumperror** [reply]
Did you find a setting to make Windsurf delay its suggestions? I couldn't seem to find one when I looked through it.

**Reply by aeckstorm** [reply]
There is no delay that I am aware of. I'm 3 days in, it does some things that I really like but it also does some of the issues that everyone is complaining about.

So far it's not as intrusive but... it still has moments. 

Copilot to me right now is not usable and is causing me more issues than anything and slowing me down which I can't have on a tight schedule. 

But I also need to have a co-helper because I am the only developer. Co-pilot was bridging that gap but right now it is useless.

**Reply by aeckstorm** [reply]
> Did you find a setting to make Windsurf delay its suggestions? I couldn't seem to find one when I looked through it.

There actually is an option to delay the response.  I also set the inline commit to shift + enter. This way i can still use tab. You can do this with copilot too.

**Comment by coredumperror** [comment]
How is a delay setting still not a thing?!? Fix this, Github!!!

**Comment by movy** [comment]
Recently copilot became unnecessary verbose and very fast to spit out a suggestion, to the point when I cannot use Tab key to indent code, as Copilot constantly offers smth and pressing Tab almost always invokes "accept unwanted suggestion" scenario. 
Whatever is happening with Copilot development behind the scenes over last week or two is not a good choice for product quality and usability.

**Reply by lovette** [reply]
I thought it was just me, but I agree with you 100%. It's so fast now I feel like I hit ESC and delete mis-accepted suggestions more than I type my own code!

**Reply by MaffooBristol** [reply]
Yeah it's too good at it. It's a weird downside of decreasing latency

I just wish I could somehow wrangle it to be a bit more chilled like it used to be. Or even just make it not suggest things if I'm mid comment.

It's funny though when it starts adding "[...] but it's pointless" or "[...] however it doesn't work" randomly as i'm typing comments, like... oi... mean...

**Comment by adnan-smlatic** [comment]
+1 for this. The completion suggestions are too aggressive, it's like trying to type code with schizophrenia.

**Comment by reynadev-oss** [comment]
Suggestion Delay – Set how long Copilot waits before showing a suggestion (default 0ms, user-configurable).

Silence After Escape – After pressing Esc to dismiss a suggestion, Copilot should stay silent for X milliseconds (user-configurable), so it doesn't instantly suggest again.

These features would reduce interruptions and make indentation or writing without suggestions easier.

**Comment by redtopia** [comment]
Seems to have gotten so much worse recently. It's getting to the point where I am hating it more than loving it, whereas a year ago it was the opposite. I like writing my own code and I don't need copilot to be so damn aggressive.

**Comment by denravonska** [comment]
I'm also contemplating disabling the plugin until we either get a delay or are able to rebind the completion to like ctrl+tab. When you have to fight the AI to be able to indent it has gone too far.

**Comment by tvillarr** [comment]
+1.  The copilot auto-completion is beginning to be insufferable to deal with.

**Comment by thomasruns** [comment]
I'm not sure what changed, but this has gotten MUCH worse lately. It's borderline unusable with how much it's constantly moving code around the screen with huge blocks of suggestions and no way to delay it.

**Comment by LzrBim** [comment]
Bump.  It's like a permanent Cumulative Layout Shift.

**Comment by ddrm86** [comment]
Second this. It is so annoying I just deactivated it.

**Comment by quangv** [comment]
any progress on this? Can't believe it's not already an option. Sometimes the code suggestions are so massive it gets in the way of manually reading / editing the code.

Could be related: https://github.com/orgs/community/discussions/9978

**Comment by ademarCardoso** [comment]
Same problem here, I hate the current status 🙃

**Comment by leonstroschein** [comment]
I'm here too because it's a big problem! Fix it please...

**Comment by prodbylcsh** [comment]
When inline suggestion was new, it felt like it was helping me code. Now with it's current aggressive suggestions that are constant and immediate, it feels like I am helping AI to code.

I don't understand how they can ignore the community for that long. They must have never used it because if they did, they would know how annoying the suggestions are with the current state. What's more annoying, those are "native" features which should always have the most user control capabilities to customize it.

**Comment by ssteinerx** [comment]
I fail to understand the delay in adding this seemingly simple, and oft duplicated request.  Maybe if someone actually combined all the similar requests instead of ignoring them, it'd be more obvious how many people are experiencing issues with this idiotically aggressive and unusable "suggesttion -- I <suggestion> don<suggestion>'t give a shi<suggestion>t what you'<suggestion>re do<suggestion>ing" feature.

**Comment by berncasa** [comment]
Bump! same here! it became much more annoying recently. it is like their goal is to start to suggest as fast as possible.
imposible to press tab to tabulate code anymore.

**Comment by kyunussov** [comment]
Please implement delay for code suggestion in copilot config

**Comment by ikkentim** [comment]
i generally have inline completion disabled because this is just too annoying to fight with all the time. this really needs to change.

**Comment by Syth-1** [comment]
+1 the vs community implementation is perfect, but the Vs code version of copilot is too aggressive,

**Comment by rsclay** [comment]
A delay would be nice, but I would really love to be able to trigger suggestions **only** when asked, like via `ctrl+space`. Disabling inline completions seems to disable it completely.

Getting suggestions as soon as you stop typing for a millisecond is distracting, and derails my train of thought, and feels like hypnosis lol.

**Reply by JulienBe** [reply]
It used to be an option to trigger the auto completion just on demand. It recently disappeared.

**Comment by redben** [comment]
Would be much appreciated. Sometimes Copilot feels like a third hand reaching out to the keyboard while you're typing...

**Comment by JoshMcCullough** [comment]
Also need this, please!

**Comment by harley-codes** [comment]
Bump !

**Comment by david-mi** [comment]
It baffles me that such an important thing hasn't been implemented yet

**Reply by ssteinerx** [reply]
You new here...I mean, like, in software dev? 😂  

Sorry...I guess I've just run out of baffles after so many inexplicable inexplicablalities.

**Reply by david-mi** [reply]
Well... I haven't been around long enough to empty my baffleometer I guess haha

**Comment by entropical** [comment]
A recent change in CoPilot in vscode has made the suggestions more intrusive, making this delay necessary.

It wasn't annoying to me, until the last week or two.

Bump!

**Reply by artalvpes** [reply]
Same thing.

I was working well without the delay but I had to disable CoPilot due to recent changes. Configurable delay is essential now!

**Reply by artalvpes** [reply]
There is an option to configure a delay after which the suggestion is accepted automatically. It is not what we need! We need a delay of inactivity after which the suggestion appears!!!

**Comment by denravonska** [comment]
@github Do you need help with this? Maybe you can ask an intern to just vibe code a delay.

**Comment by bborela** [comment]
At this point I am thinking making Copilot extremely pestering is a design choice

<img width="120" height="120" alt="image" src="https://github.com/user-attachments/assets/a7c2acaf-a174-4ea8-87b7-6a5914718b3b" />

**Comment by ubidev** [comment]
seems like a pretty efficient way to speed up free tier token pool exhaustion ;-)

**Comment by JSKoenig** [comment]
Visual Studios recently got this feature so fingers cross.

**Comment by edgardz** [comment]
This [suggestion](https://github.com/microsoft/vscode-copilot-release/issues/1013#issuecomment-3234855563) worked for me:
```
"editor.inlineSuggest.minShowDelay": 1000
```

**Reply by JoshMcCullough** [reply]
> This [suggestion](https://github.com/microsoft/vscode-copilot-release/issues/1013#issuecomment-3234855563) worked for me:
> 
> ```
> "editor.inlineSuggest.minShowDelay": 1000
> ```

This solves it, so I think this issue can be closed?

**Reply by thomasruns** [reply]
I'm not sure this is completely released yet, maybe still in beta? It does work for me, but only about 50% of the time. The rest of the time I still get immediate suggestions. At least it's an step in the right direction.

> > This [suggestion](https://github.com/microsoft/vscode-copilot-release/issues/1013#issuecomment-3234855563) worked for me:
> > ```
> > "editor.inlineSuggest.minShowDelay": 1000
> > ```
> 
> This solves it, so I think this issue can be closed?

**Reply by remyhunt** [reply]
in my editor, this is marked as an Unknown Configuration setting. additionally it doesn't change the behavior! (I'm on VSCode OSX, 1.104, latest release)

**Comment by angelo-loria** [comment]
Desperately need a delay setting on Copilot completions, had to disable it entirely for now because of how overzealous it is

**Comment by JulienBe** [comment]
Bumping this

<img width="480" height="480" alt="image" src="https://github.com/user-attachments/assets/a0256c55-34ec-4a92-93a7-b5d19833cd2c" />
